src/components/Search.js

```diff
@@ -1,34 +1,33 @@
-import React, { useState } from "react";
 import { Button, Card, CardBody, Input } from "reactstrap";
-const Search = ({ onSearchButtonClick }) => {
-  const [cityName, setCityName] = useState("");
-  const handleChange = (event) => {
-    setCityName(event.target.value);
-  };
-  const handleSubmit = () => {
-    onSearchButtonClick(cityName);
+
+const Search = ({ searchForCity, searchTerm, setSearchTerm }) => {
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    searchForCity();
   };
 
   return (
-    <Card style={{ width: "80vw" }} outline color="dark">
-      <CardBody className="d d-flex justify-content-around">
-        <Input
-          className="m-2 w-75"
-          placeholder="Enter city name to search"
-          name="cityName"
-          value={cityName}
-          onChange={handleChange}
-        />
-        <Button
-          color="success"
-          outline
-          className="fw-bold mw-auto"
-          onClick={handleSubmit}
-        >
-          Get Weather
-        </Button>
-      </CardBody>
-    </Card>
+    <form onSubmit={handleSubmit}>
+      <Card style={{ width: "80vw" }} outline color="dark">
+        <CardBody className="d d-flex justify-content-around">
+          <Input
+            className="m-2 w-75"
+            placeholder="Enter city name to search"
+            name="cityName"
+            value={searchTerm}
+            onChange={(e) => setSearchTerm(e.target.value)}
+          />
+          <Button
+            color="success"
+            outline
+            className="fw-bold mw-auto"
+            onClick={handleSubmit}
```

Comment:
this is not correct. type should be submit here. The form submit will automatically call the form onSubmit

Reply:
Changed the type to submit, removed onClick handler from the button